src/create_disk.sh

```diff
@@ -432,31 +498,16 @@ ppc64le)
     ;;
 s390x)
     bootloader_backend=zipl
-    # current zipl expects 'title' to be first line, and no blank lines in BLS file
-    # see https://github.com/ibm-s390-tools/s390-tools/issues/64
-    blsfile=$(find $rootfs/boot/loader/entries/*.conf)
-    tmpfile=$(mktemp)
-    for f in title version linux initrd options; do
-        echo $(grep $f $blsfile) >> $tmpfile
-    done
-    cat $tmpfile > $blsfile
-    # we force firstboot in building base image on s390x, ignition-dracut hook will remove
-    # this and update zipl for second boot
-    # this is only a temporary solution until we are able to do firstboot check at bootloader
-    # stage on s390x, either through zipl->grub2-emu or zipl standalone.
-    # See https://github.com/coreos/ignition-dracut/issues/84
-    # There's a similar hack in gf-set-platform
-    echo "$(grep options $blsfile | cut -d' ' -f2-) ignition.firstboot" > $tmpfile
-
-    # ideally we want to invoke zipl with bls and zipl.conf but we might need
-    # to chroot to $rootfs/ to do so. We would also do that when FCOS boot on its own.
-    # without chroot we can use --target option in zipl but it requires kernel + initramfs
-    # pair instead
-    zipl --verbose \
-        --target $rootfs/boot \
-        --image $rootfs/boot/"$(grep linux $blsfile | cut -d' ' -f2)" \
-        --ramdisk $rootfs/boot/"$(grep initrd $blsfile | cut -d' ' -f2)" \
-        --parmfile $tmpfile
+    rdcore_args=("--boot-mount=$rootfs/boot" "--kargs=ignition.firstboot")
+    if [[ ${secure_execution} -eq 1 ]]; then
+        rdcore_args+=("--secex-mode=enforce" "--rootfs=$deploy_root" "--hostkey=/dev/disk/by-id/virtio-hostkey")
+        propagate_luks_config boot
+        propagate_luks_config root
+    else
+        # in case builder itself runs with SecureExecution
+        rdcore_args+=("--secex-mode=disable")
+    fi
+    ${deploy_root}/usr/lib/dracut/modules.d/50rdcore/rdcore zipl ${rdcore_args[@]}
```

Comment:
I was randomly thinking about this PR and remembered this bit: the issue with calling `rdcore` like this is that it assumes the same userspace as the supermin environment, which isn't necessarily true (e.g. when composing RHCOS or FCOS rawhide we're dealing with very different userspaces).

So I do think we need to run this in a container as initially suggested. `bwrap` should already be in the supermin environment, or you can also use `chroot` similarly to how we do it for `coreos-installer pack osmet` in `src/osmet-pack`.

Reply:
> is it a problem?

If any of those libraries has an soname bump, or we start using a new e.g. glibc symbol (more likely via the Rust stdlib or one of our deps), the dynamic linker will fail.

Actually, that's already the case for openssl, which had an soname bump in Fedora (as part of https://fedoraproject.org/wiki/Changes/OpenSSL3.0) which RHEL8 hasn't had yet. If you try running the RHCOS `rdcore` in cosa, you'll get a linker error:

```
$ rpm2cpio coreos-installer-bootinfra-0.14.0-1.rhaos4.11.el8.x86_64.rpm | cpio -id
4628 blocks
$ usr/lib/dracut/modules.d/50rdcore/rdcore
usr/lib/dracut/modules.d/50rdcore/rdcore: error while loading shared libraries: libssl.so.1.1: cannot open shared object file: No such file or directory
```

> assuming libc has a problem, than even removed zipl call could be wrong

The removed `zipl` call is from cosa itself, and the supermin environment is derived from the cosa userspace, so it matches.